[31기 남용현] ADD: login, main 페이지 필수구현1~4 및 추가구현5~9 완료

sunnyfterrain opened the pull request:
## :: 최근 작업 주제 (하나 이상의 주제를 선택해주세요.)
- [x] 기능 추가
- [ ] 리뷰 반영
- [ ] 리팩토링
- [ ] 버그 수정
- [ ] 컨벤션 수정

<br />

## :: 구현 목표
- [Mission1] 로그인 페이지 레이아웃
- [Mission2] id, pw 입력 시 로그인 버튼 활성화 기능
- [Mission3] 메인 페이지 레이아웃
- [Mission4] 댓글 내용 입력 후 Enter press, 혹은 게시 버튼 클릭 시 댓글 추가 기능
- [Mission5] ID, PW validation
- [Mission6] 댓글 좋아요 / 삭제 기능
- [Mission7] 아이디 검색 기능
- [Mission8] nav 프로필 사진 클릭 시 메뉴 박스 생성
- [Mission9] 반응형 구현
<br />

## :: 구현 사항 설명 
1. 로그인 페이지 개발 
- id 1글자이상 & pw 1글자 이상 일시 로그인 버튼 활성화
- 아이디 '@' & 비밀번호 '5글자' 이상이 아닐시 alert 출력 유효성 검사
- id & pw 첫글자 공백 입력시 alert 출력

2. 메인 페이지 개발
- 댓글 내용 입력 후 Enter press 및 게시 버튼 클릭 시 댓글 추가
- 댓글 좋아요 및 마우스 댓글 hover시 삭제버튼 구현
- 아이디 검색기능 추가 ( 대소문자 구문 없이 검색 가능, 한글과 영어 검색가능)
- nav 프로필 사진 클릭 시 메뉴 박스 생성 및 다른 부분 클릭 시 메뉴 박스 닫힘 구현
- 반응형 구현 (브라우저 1000px 이하시 우측 레이아웃 숨김, 640px 이하시 검색창 숨김)

<br />

## :: 성장 포인트 
- 댓글 추가 시 첫글자가 공백이면 공백 제거 기능과 공백만 입력시에도 댓글이 등록되는걸 고치기 힘들었으나,
정규화 라는것을 알게되어 겨우 해결
- 검색 기능이 조금 까다로웠는데 filter와 문자열 포함여부를 공부함에 있어서 도움이 많이 되었고,
검색 결과 창 초기화가 어려웠는데 생각보다 너무 간단해서  당황했지만 많은 도움이 되었습니다.

<br />

## :: 기타 질문 및 특이 사항
- 프로필 버튼 클릭시 모달 창 열고 닫는 것을 display:none으로 하였는데 외부를 클릭 했을때 닫히는것을
구현하였을때 버블링때문에 모달창 내부를 클릭해도 닫히는 오류로 이벤트 위임으로 모든 태크가 dataset을 주었는데
코드가 너무 길어져서 다른방법이 없나 궁금합니다.

yojuyoon commented on students/18th/kimwooyoung/js/login.js:
```diff
@@ -0,0 +1,12 @@
+
+document.addEventListener('keyup', function(){
+    if( txtwrite.value.length > 0 && pwdwrite.value.length >= 6 ){
+      return btnclick.style.backgroundColor = "#0095F6";
+    }
+    else{
+      return btnclick.style.backgroundColor = "#B2DFFC";
```
자바스크립트 파일 내에서 직접 스타일 속성을 변경할 수 있지만, 기능 분리를 위하여 해당 속성을 지닌 class를 추가해주는 방식으로 접근해보는 것은 어떨까요? 아래 링크를 참고해주세요!
- [https://developer.mozilla.org/ko/docs/Web/API/Element/classList](https://developer.mozilla.org/ko/docs/Web/API/Element/classList)

yojuyoon commented on students/18th/kimwooyoung/login.html:
```diff
@@ -0,0 +1,24 @@
+  <meta charset="utf-8">
+  <meta name="viewport" content="width=device-width, initial-scale=1.0">
+  <link rel="preconnect" href="https://fonts.gstatic.com">
+  <link href="https://fonts.googleapis.com/css2?family=Lobster&display=swap" rel="stylesheet">
+  <link rel="stylesheet" href="style/login.css">
+  <link href="logo.png" rel="shortcut icon" type="image/x-icon">
+  <title>westagram</title>
```
title도 알맞게 바꿔주셨네요 👍

yojuyoon commented on students/18th/kimwooyoung/login.html:
```diff
@@ -0,0 +1,24 @@
+    <div class="login_border">
+      <input class="login_txt" type="text" placeholder="전화번호, 사용자 이름 또는 이메일">
+      <input class="login_pwd" type="password" placeholder="비밀번호">
+      <input class="login_btn" type="button" value="로그인">
```
```suggestion
      <input class="login_btn" type="button" value="로그인"/>
```
태그 제대로 닫아주세요!

yojuyoon commented on students/18th/kimwooyoung/main.html:
```diff
@@ -0,0 +1,32 @@
+            <span class="nav_title">westagram</span>
+            <input class="nav_text" type="text" placeholder="&#xf002; 검색">
+            <span class="nav_imgbox">
+                <img class="nav_img1" src="https://s3.ap-northeast-2.amazonaws.com/cdn.wecode.co.kr/bearu/explore.png" alt="search">
+                <img class="nav_img2" src="https://s3.ap-northeast-2.amazonaws.com/cdn.wecode.co.kr/bearu/heart.png" alt="heart">
+                <img class="nav_img3" src="https://s3.ap-northeast-2.amazonaws.com/cdn.wecode.co.kr/bearu/profile.png" alt="mypage">       
```
class명 또는 변수명, 함수명에 숫자, 방향을 작성하는 것은 모호한 이름이 되므로 지양해주는 것이 좋습니다. 정확한 Icon의 이름으로 변경해주세요~!

yojuyoon commented on students/18th/kimwooyoung/main.html:
```diff
@@ -0,0 +1,32 @@
+                <img class="nav_img2" src="https://s3.ap-northeast-2.amazonaws.com/cdn.wecode.co.kr/bearu/heart.png" alt="heart">
+                <img class="nav_img3" src="https://s3.ap-northeast-2.amazonaws.com/cdn.wecode.co.kr/bearu/profile.png" alt="mypage">       
+            </span>
+        </nav>
+        <main>main
+            <div class="feeds">feeds
+                <header>header</header>
+                <article>article</article>
+            </div>
+            <div class="main-right">main-right</div>
+        </main>
```
semantic tag 활용 굿!

yojuyoon commented on students/18th/kimwooyoung/style/main.css:
```diff
@@ -0,0 +1,93 @@
+    justify-content: space-around;
+}
+
+.nav_img1{
+    margin: auto 0;    
+    width: 30px;
+    height: 30px;
+}
+
+.nav_img2{
+    margin: auto 0;    
+    width: 30px;
+    height: 30px;
+}
+
+.nav_img3{
+    margin: auto 0;    
+    width: 30px;
+    height: 30px;
+}
```
동일한 속성값을 가졌네요! 중복 속성을 적용할 때에는 선택자를 다중으로 두어 한번에 속성을 할당해줄 수 있습니다.

yojuyoon commented on students/18th/kimwooyoung/style/main.css:
```diff
@@ -0,0 +1,93 @@
+}
+
+.main-right{
+    border: 1px solid gray;
```
반복되는 속성은 따로 고유한 클래스명을 가져서 속성을 할당해준 뒤 해당 클래스명을 구조에서 적용시켜줄 수 있습니다. 
```css
.border-gray {
   border: 1px solid gray;
}
```

border-gray라는 클래스명을 필요한 구조에 클래스명으로 추가해줄 수 있겠죠!

yojuyoon reviewed (commented):
우영님! 깔끔하게 구현 잘 해주셨네요! 추가 기능 구현도 화이팅 해주시고 리뷰 내용 확인해서 리팩토링도 함께 진행해주세요 :) 화이팅!

sunnyfterrain committed:
Add: 로그인 페이지 구현
- [Mission1] 로그인 페이지 레이아웃 구현
- [Mission2] id, pw입력 시 로그인 버튼 활성화 기능 구현

sunnyfterrain committed:
Add: 메인페이지 레이아웃
- [Mission3] 메인페이지 ui 레이아웃 제작 완료

sunnyfterrain committed:
Add: 댓글 기능 구현 및 유효성검사 구현
- [Mission4] 댓글 내용 입력 후 Enter 혹은 버튼 클릭시 댓글 추가 기능
- [추가구현 Mission5] ID,PW 유효성 검사 구현 (id에 @포함 및 pw 5자 이상)

sunnyfterrain committed:
Add: 댓글 좋아요/삭제 기능 추가
- [Mission6] 댓글 좋아요 및 삭제 기능 추가

sunnyfterrain committed:
Add: 아이디 검색기능, 프로필 메뉴 기능, 반응형 구현
- [Mission7] 아이디 검색기능 추가, 영어, 한글 검색기능, 대소문 구분x
- [Mission8] nav 프로필 사진 클릭시 메뉴박스 생성, 바깥화면 클릭시 닫힘
- [Mission9] 반응형 구현 브라우저 1000px 이하 시 우측 메뉴 사라짐

sunnyfterrain committed:
Fix: 프로필 버튼 클릭 이벤트 수정
- 프로필 모달 제대로 뜨지 않고 원하지 않게 창이 닫혀 수정

sunnyfterrain committed:
Modify: fetch 주소 수정 및 프로필 모달창 수정
- fetch를 못불러오는것에 대한 수정
- 프로필 모달창 외부 클릭하면 닫힘 기능 재구현
